src/meson.build

```diff
@@ -51,10 +51,14 @@ foreach h: headers
   epoxy_headers += join_paths(meson.source_root(), 'include/epoxy/@0@'.format(h))
 endforeach
 
+common_ldflags = []
 if cc.get_id() == 'gcc'
   common_ldflags = [ '-Wl,-Bsymbolic', '-Wl,-z,relro', '-Wl,-z,now', ]
```

Comment:
You're right, I don't think these options are supported on [macOS ld](http://www.manpages.info/macosx/ld.1.html).